worker/src/main/java/gov/cms/ab2d/worker/config/RoundRobinBlockingQueue.java

```diff
@@ -0,0 +1,357 @@
+package gov.cms.ab2d.worker.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
+
+@Slf4j
+public class RoundRobinBlockingQueue<E> implements BlockingQueue<E> {
+    // The list of categories
+    private List<String> categories = new ArrayList<>();
+    // The individual category queues
+    private Map<String, LinkedList<E>> categoryQueues = new HashMap<>();
+    // The current category index
+    private int currentIndex = 0;
+    // Main lock guarding all access
+    private ReentrantLock lock;
+    // Not empty condition on the lock
+    private final Condition notEmpty;
+    // Total number in the queues
+    private long count = 0;
```

Comment:
This `count` would be redundant, since you can just check the size of the queue list.